File: app/src/main/java/com/example/learningverbs/utils/Tools.java
package com.example.learningverbs.utils;


import android.content.Context;
import android.content.SharedPreferences;
import android.graphics.Bitmap;
import android.view.View;
import android.widget.Toast;

import com.example.learningverbs.utils.constants.Constants;
import com.google.android.material.snackbar.Snackbar;

import java.io.ByteArrayOutputStream;
import java.io.File;
import java.io.IOException;

import id.zelory.compressor.Compressor;


public class Tools {
    private static SharedPreferences preferences = LearningApplication.getMyApplicationContext().getSharedPreferences(Constants.PREFERENCE_FILE_KEY, Context.MODE_PRIVATE);

    public static void showSnackMessage(View myActivity, String message) {
        Snackbar.make(myActivity, message, Snackbar.LENGTH_SHORT).show();
    }

    public static void showToastMessage(String message) {
        Toast.makeText(LearningApplication.getMyApplicationContext(), message, Toast.LENGTH_SHORT).show();
    }

    public static void setStringPreference(String name, String value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(name, value);
        editor.apply();
    }

    public static String getStringPreference(String name) {
        return preferences.getString(name, "");
    }

    public static byte[] getImage(Context ctx, String path, int width, int height) {
        final File file_thumb_path = new File(path);

        Bitmap thumb_bitmap = null;
        byte[] thumb_byte = null;
        try {
            thumb_bitmap = new Compressor(ctx)
                    .setMaxWidth(width)
                    .setMaxHeight(height)
                    .setQuality(75)
                    .compressToBitmap(file_thumb_path);
        } catch (IOException e) {
            e.printStackTrace();
        }
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        thumb_bitmap.compress(Bitmap.CompressFormat.JPEG, 80, baos);
        thumb_byte = baos.toByteArray();


        return thumb_byte;

    }
}

File: app/src/main/java/com/karla/learningverbs/kotlin/utils/Tools.kt
package com.karla.learningverbs.utils

import android.content.Context
import android.graphics.Bitmap
import android.view.View
import android.widget.Toast
import com.google.android.material.snackbar.Snackbar
import com.karla.learningverbs.kotlin.utils.constants.Constants
import id.zelory.compressor.Compressor
import java.io.ByteArrayOutputStream
import java.io.File
import java.io.IOException

object Tools {
    private val preferences = LearningApplication2.getMyApplicationContext().getSharedPreferences(
        Constants.PREFERENCE_FILE_KEY, Context.MODE_PRIVATE
    )

    fun showSnackMessage(myActivity: View?, message: String?) {
        Snackbar.make(myActivity!!, message!!, Snackbar.LENGTH_SHORT).show()
    }

    fun showToastMessage(message: String?) {
        Toast.makeText(LearningApplication2.getMyApplicationContext(), message, Toast.LENGTH_SHORT)
            .show()
    }

    fun setStringPreference(name: String?, value: String?) {
        val editor = preferences.edit()
        editor.putString(name, value)
        editor.apply()
    }

    fun getStringPreference(name: String?): String? {
        return preferences.getString(name, "")
    }

    fun getImage(ctx: Context?, path: String?, width: Int, height: Int): ByteArray? {
        val file_thumb_path = File(path)
        var thumb_bitmap: Bitmap? = null
        var thumb_byte: ByteArray? = null
        try {
            thumb_bitmap = Compressor(ctx)
                .setMaxWidth(width)
                .setMaxHeight(height)
                .setQuality(75)
                .compressToBitmap(file_thumb_path)
        } catch (e: IOException) {
            e.printStackTrace()
        }
        val baos = ByteArrayOutputStream()
        thumb_bitmap!!.compress(Bitmap.CompressFormat.JPEG, 80, baos)
        thumb_byte = baos.toByteArray()
        return thumb_byte
    }
}